src/main/scala/com/fulcrumgenomics/umi/ExtractUmisFromBam.scala

```diff
@@ -191,9 +193,44 @@ object ExtractUmisFromBam {
           case _ => None
         }
     }
+    // update any clipping information
+    updateClippingInformation(record=record, clippingAttribute=clippingAttribute, readStructureBases=readStructureBases)
     record.setReadString(basesAndQualities.map(_._1).mkString)
     record.setBaseQualityString(basesAndQualities.map(_._2).mkString)
     // return the concatenation of the molecular barcodes in sequencing order
     molecularBarcodeBases.mkString
   }
+
+
+  /**
+    * Update the clipping information produced by Picard's MarkIlluminaAdapters to account for any non-template bases
+    * that will be removed from the read.
+    */
+  private[umi] def updateClippingInformation(record: SAMRecord,
+                                             clippingAttribute: Option[String],
+                                             readStructureBases: Seq[(String, String, ReadSegment)]): Unit = {
+    clippingAttribute.map(tag => (tag, record.getIntegerAttribute(tag))) match {
+      case None =>
```

Comment:
I think it's nice to put `Unit` on the right hand side of these cases to indicate clear intention of doing nothing.